content/docs/reconciliation.md

```diff
@@ -128,30 +128,29 @@ In order to solve this issue, React supports a `key` attribute. When children ha
+現在 React 知道只有帶著 `'2014'` key 的元素是新元素，而帶著 `'2015'` 以及 `'2016'` key 的元素只是相對位置移動了。
 
-In practice, finding a key is usually not hard. The element you are going to display may already have a unique ID, so the key can just come from your data:
+在現實中，產生一個 key 並不困難。你要展現的元素可能已經有了一個可以唯一識別的 ID，於是 key 可以直接從你的數據中提取：
 
 ```js
 <li key={item.id}>{item.name}</li>
 ```
 
-When that's not the case, you can add a new ID property to your model or hash some parts of the content to generate a key. The key only has to be unique among its siblings, not globally unique.
+當以上情況不成立時，你可以新增一個 ID 字串到你的 Model 中，或者利用一部分內容作為哈希值 (hash) 來生成一個 key。這個 key 不需要全局唯一，但在陣列中需要保持唯一。
 
-As a last resort, you can pass an item's index in the array as a key. This can work well if the items are never reordered, but reorders will be slow.
+最後，你也可以使用元素在陣列中的索引值作為 key。這個方法在元素不進行重新排序時比較合適，但如果有順序修改，diff 就會變慢。
 
-Reorders can also cause issues with component state when indexes are used as keys. Component instances are updated and reused based on their key. If the key is an index, moving an item changes it. As a result, component state for things like uncontrolled inputs can get mixed up and updated in unexpected ways.
+當使用陣列索引值作為 key 的 Component 進行重新排序時， Component state 可能會遇到一些問題。由於 Component 實例是基於它們的 key 來決定是否更新以及重複使用，如果 key 是一個索引值，那麼修改順序時會修改當前的 key，導致 Component 的 state（例如不受控制輸入框）可能相互篡改導致無法預期的變動。
 
-[Here](codepen://reconciliation/index-used-as-key) is an example of the issues that can be caused by using indexes as keys on CodePen, and [here](codepen://reconciliation/no-index-used-as-key) is an updated version of the same example showing how not using indexes as keys will fix these reordering, sorting, and prepending issues.
+[這是](codepen://reconciliation/index-used-as-key) 在 Codepen 上的例子，展示使用索引值作為 key 時導致的問題，以及 [這裡](codepen://reconciliation/no-index-used-as-key) 是一個不使用索引值作為 key 的例子的版本，修復了重新排列，排序，以及在陣列首插入的問題。
 
-## Tradeoffs {#tradeoffs}
+## 權衡 {#tradeoffs}
 
-It is important to remember that the reconciliation algorithm is an implementation detail. React could rerender the whole app on every action; the end result would be the same. Just to be clear, rerender in this context means calling `render` for all components, it doesn't mean React will unmount and remount them. It will only apply the differences following the rules stated in the previous sections.
+請記得 Reconciliation 只是一個實現細節，即使 React 在每個動作後之後對整個應用進行重新渲染，得到的最終結果也會和進行 Reconciliation 後是一樣的。在這個情形下，重新渲染表示在所有 Component 內呼叫 `render`，這不代表 React 會卸載或裝載它們。React 只會基於以上提到的規則來決定如何進行差異的合併。
 
-We are regularly refining the heuristics in order to make common use cases faster. In the current implementation, you can express the fact that a subtree has been moved amongst its siblings, but you cannot tell that it has moved somewhere else. The algorithm will rerender that full subtree.
+我們會定期優化演算法，讓常見的使用場景能夠更有效率地執行。在當前的實現中，可以理解為一棵子樹能在其相同階層的兄弟之間移動，但不能移動到其他位置。如果移動到其他位置，則演算法會重新渲染整棵子樹。
```

Comment:
```suggestion
我們會定期最佳化啟發式演算法，讓常見的使用場景能夠更有效率地執行。在目前的實作中，可以理解為一棵 subtree 能在其相同階層的兄弟之間移動，但不能移動到其他位置。如果移動到其他位置，則演算法會重新 render 整棵 subtree。
```

Reply:
Here I think using 優化 is more appropriate for refining.

Reply:
How about `我們會定期改善啟發式演算法`？

Reply:
That will be great. Just wondering why `優化` doesn't suit the situation here. 
I'm a newbie to document translation, so I'd like to know the thoughts behind to avoid making the same mistake next time 😅. Thanks!

Reply:
It's my habit just like `建立` better than `創建`. I want to use more accurate terms in the document instead of use coming from Simplified Chinese terms.

Reply:
I see. Thanks for sharing that. 
I have amended those changes to this commit.